controller/job/backresthandler.go

```diff
@@ -0,0 +1,229 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"fmt"
+	"time"
+
+	crv1 "github.com/crunchydata/postgres-operator/apis/cr/v1"
+	"github.com/crunchydata/postgres-operator/config"
+	"github.com/crunchydata/postgres-operator/controller"
+	"github.com/crunchydata/postgres-operator/events"
+	"github.com/crunchydata/postgres-operator/kubeapi"
+	backrestoperator "github.com/crunchydata/postgres-operator/operator/backrest"
+	clusteroperator "github.com/crunchydata/postgres-operator/operator/cluster"
+	"github.com/crunchydata/postgres-operator/util"
+	log "github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/batch/v1"
+)
+
+// backrestUpdateHandler is responsible for handling updates to backrest jobs
+func (c *Controller) handleBackrestUpdate(job *apiv1.Job) error {
+
+	// return if job wasn't successful
+	if !isJobSuccessful(job) {
+		log.Debugf("jobController onUpdate job %s was unsuccessful and will be ignored",
+			job.Name)
+		return nil
+	}
+
+	// return if job is being deleted
+	if isJobInForegroundDeletion(job) {
+		log.Debugf("jobController onUpdate job %s is being deleted and will be ignored",
+			job.Name)
+		return nil
+	}
+
+	labels := job.GetObjectMeta().GetLabels()
+
+	// Route the backrest job update to the appropriate function depending on the type of
+	// job.  Please note that thee LABE_PGO_CLONE_STEP_2 label represents a special form of
+	// pgBackRest restore that is utilized as part of the clone process.  Since jobs with
+	// the LABEL_PGO_CLONE_STEP_2 also inlcude the LABEL_BACKREST_RESTORE label, it is
+	// necessary to first check for the presence of the LABEL_PGO_CLONE_STEP_2 prior to the
+	// LABEL_BACKREST_RESTORE label to determine if the restore is part of and ongoing clone.
+	switch {
+	case labels[config.LABEL_BACKREST_COMMAND] == "backup":
+		c.handleBackrestBackupUpdate(job)
+	case labels[config.LABEL_PGO_CLONE_STEP_2] == "true":
+		c.handleCloneBackrestRestoreUpdate(job)
+	case labels[config.LABEL_BACKREST_RESTORE] == "true":
+		c.handleBackrestRestoreUpdate(job)
+	case labels[config.LABEL_BACKREST_COMMAND] == crv1.PgtaskBackrestStanzaCreate:
+		c.handleBackrestStanzaCreateUpdate(job)
+	}
+
+	return nil
+}
+
+// handleBackrestRestoreUpdate is responsible for handling updates to backrest restore,
+// (specifically those that have not been submitted as part of the clone process)
+func (c *Controller) handleBackrestRestoreUpdate(job *apiv1.Job) error {
+
+	labels := job.GetObjectMeta().GetLabels()
+	log.Debugf("jobController onUpdate backrest restore job case")
+	log.Debugf("got a backrest restore job status=%d", job.Status.Succeeded)
+	log.Debugf("set status to restore job completed  for %s", labels[config.LABEL_PG_CLUSTER])
+	log.Debugf("workflow to update is %s", labels[crv1.PgtaskWorkflowID])
+	if err := util.Patch(c.JobClient, patchURL, crv1.JobCompletedStatus, patchResource, job.Name,
```

Comment:
a kind request for a newline between the debug and conditional

Reply:
Added some spacing.